easybuild/tools/modules.py

```diff
@@ -718,6 +719,53 @@ def loaded_modules(self):
 
         return loaded_modules
 
+    def check_loaded_modules(self):
+        """
+        Check whether any (EasyBuild-generated) modules are loaded already in the current session
+        """
+        ignored_keys = [get_software_root_env_var_name(x) for x in build_option('allow_loaded_modules') or []]
+
+        eb_module_keys = []
+        for key in os.environ:
+            if key.startswith(ROOT_ENV_VAR_NAME_PREFIX) and key not in ignored_keys:
+                eb_module_keys.append(key)
+
+        if eb_module_keys:
+            loaded_modules = self.loaded_modules()
+
+            # try to track down modules that define the $EBROOT* environment variables that were found
+            loaded_eb_modules = []
+            for loaded_module in loaded_modules:
+                out = self.show(loaded_module)
+                for key in eb_module_keys[:]:
+                    if key in out:
+                        loaded_eb_modules.append(loaded_module)
+                        eb_module_keys.remove(key)
+
+            # warn about $EBROOT* environment variables without matching loaded module
+            if eb_module_keys:
+                tup = (len(eb_module_keys), ROOT_ENV_VAR_NAME_PREFIX, '$' + ', $'.join(eb_module_keys))
+                print_warning("Found %d defined $%s* environment variables without matching loaded module: %s" % tup)
```

Comment:
I guess it simply depends on how you want to treat "may be harmless" vs "won't be harmless" vs "will be harmless". It looks safer to me to err on the safe side and decline to proceed it is is not sure that it will be harmless. But in any case, I have no strong objection against keeping it as warning.